File: BankKata/BankCore/Models/Clock.cs
using System;

namespace BankCore.Models
{
    public class Clock
    {
        private const string dd_MM_yyyy = "dd/MM/yyyy";

        public virtual string TodayAsString()
        {
            return Today();
        }

        protected virtual string Today()
        {
            return DateTime.Today.ToString(dd_MM_yyyy);
        }
    }
}

File: BankKata/BankCoreVBNetEdition/Clock.vb
Public Class Clock
    Private Const DdMmYyyy As String = "dd/MM/yyyy"

    Public Overridable Function TodayAsString() As String
        Return Today()
    End Function

    Protected Overridable Function Today() As String
        Return Date.Today.ToString(DdMmYyyy)
    End Function
End Class